### DISCUSSION TITLE
SHA512 file hasher much slower than MD5 file hasher

### DISCUSSION THREAD

**Discussion by Hayababuza** [Daninet/hash-wasm#39] [Q&A] [answered]
Hello,

I'm new in this field and I'm trying to insert the file hasher like it was suggested by Daninet in stackoverflow: https://stackoverflow.com/a/63287199/6251760 (Link copied from the README.md file) but instead of using a MD5 hash instance I would like to use the SHA512 instance.
Everything works fine with one exception, the SHA512 instance is much slower than the MD5 instance when updating the hasher.

For example I tried to hash a file with 1.4 Gb:
- SHA512 hasher needed about 66 seconds in total and about 20 seconds/chunk
- MD5 hasher needed about 16 seconds in total and about 5 seconds/chunk

Is it possible that I have to write it differently for SHA512 or are these numbers correct?

**Comment by Daninet** [comment, marked as answer]
This is normal. The SHA512 algorithm is more secure but also more expensive from CPU perspective. (it has 64 rounds compared to MD5's 4 rounds). If you need a fast hashing algorithm and you don't need security, you could consider using XXHash32/64.

**Reply by Hayababuza** [reply]
@Daninet 
I have another question about the hash time.

How come it was supposed to hash 400Mb/s according to your example but took 16 seconds for a 1.4GB file? Shouldn't it take 3.5 seconds or did I misunderstand something?
